webpack/assets/javascripts/react_app/components/ForemanModal/__tests__/ForemanModal.integration.test.js

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import ForemanModal, { reducers } from '../index';
+import ForemanModalHeader from '../subcomponents/ForemanModalHeader';
+import ForemanModalFooter from '../subcomponents/ForemanModalFooter';
+import IntegrationTestHelper from '../../../common/IntegrationTestHelper';
+
+import { setModalOpen, setModalClosed, addModal } from '../ForemanModalActions';
+
+// This file is for integration tests of the Redux-connected ForemanModal component
+
+describe('ForemanModal - integration tests', () => {
+  it('should add, open, and close modals as directed by Redux actions', () => {
+    const integrationTestHelper = new IntegrationTestHelper(reducers);
+
+    integrationTestHelper.takeStoreSnapshot('initial state');
+
+    integrationTestHelper.store.dispatch(addModal({ id: 'modal1' }));
+    integrationTestHelper.store.dispatch(addModal({ id: 'modal2' }));
+    integrationTestHelper.store.dispatch(addModal({ id: 'modal3' }));
+
+    const modal1 = integrationTestHelper.mount(
+      <ForemanModal id="modal1" title="modal1 title" />
+    );
+    const modal2 = integrationTestHelper.mount(
+      <ForemanModal id="modal2" title="modal1 title" />
+    );
+    const modal3 = integrationTestHelper.mount(
+      <ForemanModal id="modal3" title="modal1 title" />
+    );
+
+    // After a Redux action updates the component, this ensures we're looking at
+    // the latest version of the React tree after rerender.
+    const updateWrappers = () =>
```

Comment:
Using `await IntegrationTestHelper.flushAllPromises();` instead `updateWrappers` would work?

Reply:
@sharvit I tried it but the test failed.